vm/systemSmartContracts/auction_test.go

```diff
@@ -1151,6 +1301,219 @@ func TestAuctionStakingSC_ExecuteStakeNotEnoughArgsShouldErr(t *testing.T) {
+	arguments := CreateVmContractCallInput()
+	arguments.Function = "stake"
+	eei.SetGasProvided(arguments.GasProvided)
+	arguments.CallValue = big.NewInt(15)
+	arguments.Arguments = [][]byte{
+		big.NewInt(2).Bytes(),
+		stakerPubKey1.Bytes(), []byte("signed"),
+		stakerPubKey2.Bytes(), []byte("signed"),
+	}
+
+	retCode := stakingSmartContract.Execute(arguments)
+	assert.Equal(t, vmcommon.OutOfFunds, retCode)
+
+	balance := eei.GetBalance(arguments.CallerAddr)
+	assert.Equal(t, big.NewInt(0), balance)
+}
+
+func TestAuctionStakingSC_ExecuteStakeNotEnoughGasForMultipleNodesShouldErr(t *testing.T) {
+	t.Parallel()
+	stakerPubKey1 := big.NewInt(101)
+	stakerPubKey2 := big.NewInt(102)
+	args := createMockArgumentsForAuction()
+
+	blockChainHook := &mock.BlockChainHookStub{
+		GetUserAccountCalled: func(address []byte) (vmcommon.UserAccountHandler, error) {
+			return nil, state.ErrAccNotFound
+		},
+	}
+	eei, _ := NewVMContext(blockChainHook, hooks.NewVMCryptoHook(), parsers.NewCallArgsParser(), &mock.AccountsStub{}, &mock.RaterMock{})
+
+	argsStaking := createMockStakingScArguments()
+	argsStaking.StakingSCConfig = args.StakingSCConfig
+	argsStaking.Eei = eei
+	stakingSC, _ := NewStakingSmartContract(argsStaking)
+
+	eei.SetSCAddress([]byte("addr"))
+	_ = eei.SetSystemSCContainer(&mock.SystemSCContainerStub{GetCalled: func(key []byte) (contract vm.SystemSmartContract, err error) {
+		return stakingSC, nil
+	}})
+
+	args.Eei = eei
+	args.StakingSCConfig.GenesisNodePrice = "10"
+	args.GasCost.MetaChainSystemSCsCost.Stake = 10
+	stakingSmartContract, _ := NewStakingAuctionSmartContract(args)
+	arguments := CreateVmContractCallInput()
+	arguments.Function = "stake"
+	arguments.GasProvided = 15
+	eei.SetGasProvided(arguments.GasProvided)
+	arguments.CallValue = big.NewInt(15)
+	arguments.Arguments = [][]byte{
+		big.NewInt(2).Bytes(),
+		stakerPubKey1.Bytes(), []byte("signed"),
+		stakerPubKey2.Bytes(), []byte("signed"),
+	}
+
+	retCode := stakingSmartContract.Execute(arguments)
+	assert.Equal(t, vmcommon.OutOfGas, retCode)
+
+	balance := eei.GetBalance(arguments.CallerAddr)
+	assert.Equal(t, big.NewInt(0), balance)
+}
+
+func TestAuctionStakingSC_ExecuteStakeOneKeyFailsOnStakeSCShouldStakeForOnlyOne(t *testing.T) {
+	t.Parallel()
+
+	args := createMockArgumentsForAuction()
+
+	stakerAddress := big.NewInt(100)
+	stakerPubKey := big.NewInt(100)
+
+	blockChainHook := &mock.BlockChainHookStub{}
+	nodePrice, _ := big.NewInt(0).SetString(args.StakingSCConfig.GenesisNodePrice, 10)
+	expectedRegistrationData := AuctionData{
+		RewardAddress:   stakerAddress.Bytes(),
+		RegisterNonce:   0,
+		Epoch:           0,
+		BlsPubKeys:      [][]byte{stakerPubKey.Bytes()},
+		TotalStakeValue: big.NewInt(0).Mul(nodePrice, big.NewInt(2)),
+		LockedStake:     big.NewInt(0),
+		MaxStakePerNode: big.NewInt(0).Mul(nodePrice, big.NewInt(2)),
+		//TODO: why num registered 0? 1 could be registered
+		NumRegistered: 0,
```

Comment:
this should be 1. but depends on what the mock stakingSC returns. It would be better to use a non mocked stakingSC.

Reply:
fixed